app/javascript/containers/ReviewContainer.js

```diff
@@ -0,0 +1,87 @@
+import React, { Component } from "react"
+import ReviewEditFormTile from "../tiles/ReviewEditFormTile";
+import ReviewTile from "../tiles/ReviewTile"
+
+
+
+class ReviewContainer extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      edit: false,
+      review: {}
+    };
+    this.handleEditClick = this.handleEditClick.bind(this)
+    this.handleSubmit = this.handleSubmit.bind(this)
+    this.updateReview = this.updateReview.bind(this)
+    this.handleChange = this.handleChange.bind(this)
+  }
+  handleEditClick(){
+    this.setState({edit: true})
+  }
+
+  handleChange(event){
+    let review = this.state.review
+    review[event.target.name] = event.target.value
+    this.setState({review: review})
+  }
+
+  handleSubmit() {
+    let review = { review: this.state.review}
+    this.updateReview(review)
+  }
+
+
+  updateReview(review) {
+    fetch(`/api/v1/reviews/${this.state.review.id}`, {
+      method: 'PATCH',
+      body: JSON.stringify(review),
+      credentials: 'same-origin',
+      headers: {
+        'Accept': 'application/json',
+        'Content-Type': 'application/json'
+      }
+    })
+    .then(response => {
+      if (response.ok) {
+        return response
+      } else {
+        let errorMessage = `${response.status} (${response.statusText})`, error = new Error(errorMessage)
+        throw (error)
+      }
+    })
+    .then(response => response.json())
+    .then(body => {
+      let rev = this.state.review
+      rev.rating = body.rating
+      rev.comment = body.comment
+      this.setState({review: rev, edit: false})
+    })
+    .catch(error => {
+      console.error(`Error in fetch: ${error.message}`),
+      alert("there was a problem with the submission")
+    });
+  }
+
+  componentDidMount(){
+    this.setState({review: this.props.review})
+  }
+
+
+  render() {
+    let review =()=>{
+      if(this.state.edit){
+        return <ReviewEditFormTile handleSubmit={this.handleSubmit} review={this.state.review} handleChange={this.handleChange}/>
+      } else{
+        return <ReviewTile review={this.state.review} onClick={this.handleEditClick} />
+      }
+    }
```

Comment:
this probably doesnt need to be a function. Just make it a `review` variable.

Reply:
We made this into a function to be DRYer.